lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -351,20 +351,103 @@ test.serial('migrateGranuleRecord handles nullable fields on source granule data
   );
 });
 
-test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error if previously migrated record is newer', async (t) => {
   const {
     knex,
     testGranule,
   } = t.context;
 
-  await migrateGranuleRecord(testGranule, knex);
+  const testGranule1 = testGranule;
+  const testGranule2 = {
+    ...testGranule1,
+    updatedAt: Date.now() - 1000,
+  };
+
+  await migrateGranuleRecord(testGranule1, knex);
 
   await t.throwsAsync(
-    migrateGranuleRecord(testGranule, knex),
+    migrateGranuleRecord(testGranule2, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error the record was previously migrated and the new record is "running"', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testGranule1 = testGranule;
+  const testGranule2 = {
+    ...testGranule1,
+    status: 'running',
+    updatedAt: Date.now(),
+  };
+
+  await migrateGranuleRecord(testGranule1, knex);
+
+  await t.throwsAsync(
+    migrateGranuleRecord(testGranule2, knex),
     { instanceOf: RecordAlreadyMigrated }
   );
 });
 
+test.serial('migrateGranuleRecord updates an already migrated record if the updated date is newer', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    testCollection,
+    testExecution,
+    collectionCumulusId,
+  } = t.context;
+
+  const testGranule = generateTestGranule({
```

Comment:
I would explicitly set `updatedAt` on this first granule to something that is older than the `Date.now()` timestamp on the `newerGranule`. Obviously what you have here works but it's preferable to be a bit more explicit in the test setup

Reply:
👍 https://github.com/nasa/cumulus/pull/2162/commits/5ee1061cf1b4832853e78a61cbbe3f0acfe0bbb7